### PR TITLE
feat(kuma-cp): deprecated `from[].targetRef.kind: MeshService`

### PR DESCRIPTION
**Pull request by lobkovilya**
### Checklist prior to review

- [X] [Link to relevant issue][1] as well as docs and UI issues -- Fix https://github.com/kumahq/kuma/issues/9258
- [X] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: app/kumactl/cmd/apply/apply.go [hunk 1]
```diff
@@ -153,21 +160,29 @@ $ kumactl apply -f https://example.com/resource.yaml
...
+			return nil, err
 		}
 	}
 	if err := newRes.SetSpec(res.GetSpec()); err != nil {
-		return err
+		return nil, err
 	}
-	return rs.Update(ctx, newRes, store.UpdateWithLabels(meta.GetLabels()))
+	uerr := rs.Update(warnContext, newRes, store.UpdateWithLabels(meta.GetLabels()))
+	return warnings, uerr
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/apply/apply.go:187]: we really need to introduce proper client. It's quite a big workaround. Ofc not a subject of this PR

File: pkg/core/resources/model/resource.go [hunk 2]
```diff
@@ -128,6 +128,13 @@ func Validate(resource Resource) error {
 	return nil
 }
 
+func CheckDeprecated(resource Resource) []string {
```
Comment (jakubdyszkiewicz) [inline pkg/core/resources/model/resource.go:131]: nit: For me `check` should result in true/false or error/not error. I'd name it maybe `Deprecations()`

**Review by jakubdyszkiewicz — APPROVED** [review]

File: app/kumactl/cmd/apply/apply.go [hunk 3]
```diff
@@ -138,9 +139,15 @@ $ kumactl apply -f https://example.com/resource.yaml
 						return err
 					}
 				} else {
-					if err := upsert(cmd.Context(), pctx.Runtime.Registry, rs, resource); err != nil {
+					warnings, err := upsert(cmd.Context(), pctx.Runtime.Registry, rs, resource)
+					if err != nil {
 						return err
 					}
+					for _, w := range warnings {
+						if _, err := fmt.Fprintf(cmd.ErrOrStderr(), "Warning: %v\n", w); err != nil {
```
Comment (lahabana) [inline app/kumactl/cmd/apply/apply.go:147]: Worth adding a `#` ?

File: app/kumactl/cmd/apply/apply.go (hunk 1 above)
Comment (lahabana) [reply app/kumactl/cmd/apply/apply.go:187]: Yeah we need to finish the openAPI spec and just autogen most of the client!

**Comment by lahabana** [comment]
FaultInjection is the only one that allows MeshService in from?

**Comment by lobkovilya** [comment]
> FaultInjection is the only one that allows MeshService in from?

There are 2 – MeshFaultInjection and MeshTrafficPermission

File: app/kumactl/cmd/apply/apply.go (hunk 3 above)
Comment (lobkovilya) [reply app/kumactl/cmd/apply/apply.go:147]: sorry not sure I understand, you mean adding `#` somewhere in the output line?

Comment (lahabana) [reply app/kumactl/cmd/apply/apply.go:147]: That way if we return a yaml the warning is a comment. Though this is Stderr so it's fine

File: pkg/core/resources/model/resource.go [hunk 4]
```diff
@@ -128,6 +128,13 @@ func Validate(resource Resource) error {
 	return nil
 }
 
+func Deprecations(resource Resource) []string {
+	if v, ok := interface{}(resource).(interface{ Deprecated() []string }); ok {
```
Comment (lahabana) [inline pkg/core/resources/model/resource.go:132]: `GetDeprecations()` or just `Deprecations()` might be a better name here no?

Comment (lobkovilya) [reply pkg/core/resources/model/resource.go]: Right, `Deprecations` sounds better

**Review by lukidzi — APPROVED** [review]

**Pull request merged**